### PR TITLE
Council branch

### PR DESCRIPTION
**Pull request by prateekcode**
CouncilLayout Layout has been fixed.
I have used Recyclerview to do this.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
       <configuration PROFILE_NAME="Debug" CONFIG_NAME="Debug" />
     </configurations>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (sonusourav) [inline .idea/misc.xml]: Use JDK_1_8 only.

File: app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilActivity.java [hunk 2]
```diff
@@ -58,22 +69,47 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        FrSportsSecy.setOnClickListener(this);
-        EmergencyContacts.setOnClickListener(this);
+        recyclerView = findViewById(R.id.recycler_view_council);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new GridLayoutManager(this, numberOfColumns));
+        recyclerAdapter = new CouncilRecyclerAdapter(mCouncilName, mCouncilImage);
+        recyclerView.setAdapter(recyclerAdapter);
 
 
+
```
Comment (sonusourav) [inline app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilActivity.java]: Remove this extra line.

File: app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilActivity.java [hunk 3]
```diff
@@ -58,22 +69,47 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        mCouncilImage.add(R.drawable.meeting);
+        mCouncilImage.add(R.drawable.sports);
+        mCouncilImage.add(R.drawable.office_worker);
+        mCouncilImage.add(R.drawable.food);
+        mCouncilImage.add(R.drawable.boy);
+        mCouncilImage.add(R.drawable.sport_junior);
+        mCouncilImage.add(R.drawable.music_girl);
+        mCouncilImage.add(R.drawable.emergency_call);
+    }
+
```
Comment (sonusourav) [inline app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilActivity.java:86]: Remove this line also and similarly in other places.

File: app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilRecyclerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
```
Comment (sonusourav) [inline app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilRecyclerAdapter.java:12]: Run Optimise the imports option in android studio.

File: app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilRecyclerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.ArrayList;
+
+public class CouncilRecyclerAdapter extends RecyclerView.Adapter<CouncilRecyclerAdapter.ItemViewHolder> {
+
+    private ArrayList<String> mCouncilName;
+    private ArrayList<Integer> mCouncilImage;
+
+    CouncilRecyclerAdapter(ArrayList<String> mCouncilName, ArrayList<Integer> mCouncilImage) {
+        this.mCouncilName = mCouncilName;
```
Comment (sonusourav) [inline app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilRecyclerAdapter.java]: Don't use `Hungarian notation`. [Reason](https://stackoverflow.com/questions/111933/why-shouldnt-i-use-hungarian-notation)

File: app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilRecyclerAdapter.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        this.mCouncilImage = mCouncilImage;
+    }
+
+    @NonNull
+    @Override
+    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View view = inflater.inflate(R.layout.council_single_item, parent, false);
```
Comment (sonusourav) [inline app/src/main/java/com/iitdh/sonusourav/instigo/Council/CouncilRecyclerAdapter.java]: These two lines can be made into one. Don't create unnecessary variables.

File: app/src/main/res/layout/council_single_item.xml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        android:ellipsize="none"
+        android:textStyle="bold"
+        android:background="#A43F51B5"
+        android:layout_gravity="bottom"
+        android:textSize="12sp"
+        android:maxLines="3"
+        android:layout_width="match_parent"
+        android:layout_height="35dp"/>
+
+
```
Comment (sonusourav) [inline app/src/main/res/layout/council_single_item.xml]: Remove extra lines here also.

**Review by sonusourav — CHANGES_REQUESTED** [review]
Compress the images before pushing. Use [this](https://www.iloveimg.com/resize-image)

**Comment by prateekcode** [comment]
As after your suggestion I fixed the following issues:
1. Compressed Image
2. Used JDK 1_8
3. Removed all extra lines
4. Optimized Imports
5. Not using Hungarian Notation
6. Not using unnecessary variables

**Review by sonusourav — APPROVED** [review]

**Pull request merged**